conversational-client/src/app/data/conversation.ts

```diff
@@ -7,3 +7,14 @@ export interface ChatMessage {
 // Controls the agent used in a conversation.
 export const AGENT_IDS = ['default', 'fake', 'history_tutor'] as const;
 export type AgentId = (typeof AGENT_IDS)[number];
+
+// Represents a stateful conversational agent.
+export interface Agent {
+  updateState(state: AgentState): AgentState;
```

Comment:
shall we rename this to chat(state: AgentState) since people might want to do something other than updateState in this call & it makes it nicely symmetrical with the server.

we could get fancy and have

onUserMessage
onAIMessage

in case implementors want to do something *before* sending to the serve and *after* ?
I don't have a use case for that yet but it is nicely explicit & obviously linked to the UI interaction?

wdyt?